License issue

See https://github.com/conda-forge/conda-forge.github.io/issues/1052

cc @conda-forge/staged-recipes, @ocefpaf

@isruf, theoretically, it is possible to do so with:

```sh
$ go-licenses save github.com/restic/restic/cmd/restic --save_path=licenses
```

However, I'm currently encountering an issue that is stopping the program from working properly:

```
F1122 08:56:39.413984   90131 main.go:43] errors for ["github.com/restic/restic/cmd/restic"]:
github.com/cpuguy83/go-md2man/v2/md2man: /Users/andre/go/pkg/mod/github.com/spf13/cobra@v1.2.1/doc/man_docs.go:27:2: missing go.sum entry needed to verify package github.com/cpuguy83/go-md2man/v2/md2man (imported by github.com/spf13/cobra/doc) is provided by exactly one module; to add:
	go get github.com/spf13/cobra/doc@v1.2.1
```

If all works fine, it should generate a directory called `licenses` with every license from every project used.  However, as seen from CSV output, this is going to be very repetitive.

@jakirkham: as for packaging `go-licenses`,  that may be a good thing to have in the base software environment.  How do you envisage having this routinely done in packages?  Manually at every release?  Automatically with a bot update?

--- a/.azure-pipelines/azure-pipelines-linux.yml
+++ b/.azure-pipelines/azure-pipelines-linux.yml
@@ -11,7 +11,7 @@ jobs:
       linux_64_:
         CONFIG: linux_64_
         UPLOAD_PACKAGES: 'True'
-        DOCKER_IMAGE: quay.io/condaforge/linux-anvil-comp7
+        DOCKER_IMAGE: quay.io/condaforge/linux-anvil-cos7-x86_64
   timeoutInMinutes: 360
 
   steps:

--- a/.azure-pipelines/azure-pipelines-win.yml
+++ b/.azure-pipelines/azure-pipelines-win.yml
@@ -5,7 +5,7 @@
 jobs:
 - job: win
   pool:
-    vmImage: vs2017-win2016
+    vmImage: windows-2019
   strategy:
     matrix:
       win_64_:

--- a/.ci_support/linux_64_.yaml
+++ b/.ci_support/linux_64_.yaml
@@ -5,11 +5,8 @@ channel_sources:
 channel_targets:
 - conda-forge main
 docker_image:
-- quay.io/condaforge/linux-anvil-comp7
+- quay.io/condaforge/linux-anvil-cos7-x86_64
 go_compiler:
 - go-nocgo
 target_platform:
 - linux-64
-zip_keys:
-- - cdt_name
-  - docker_image

--- a/.scripts/build_steps.sh
+++ b/.scripts/build_steps.sh
@@ -5,6 +5,8 @@
 # changes to this script, consider a proposal to conda-smithy so that other feedstocks can also
 # benefit from the improvement.
 
+# -*- mode: jinja-shell -*-
+
 set -xeuo pipefail
 export FEEDSTOCK_ROOT="${FEEDSTOCK_ROOT:-/home/conda/feedstock_root}"
 source ${FEEDSTOCK_ROOT}/.scripts/logging_utils.sh
@@ -25,10 +27,10 @@ conda-build:
  root-dir: ${FEEDSTOCK_ROOT}/build_artifacts
 
 CONDARC
-GET_BOA=boa
-BUILD_CMD=mambabuild
 
-conda install --yes --quiet "conda-forge-ci-setup=3" conda-build pip ${GET_BOA:-} -c conda-forge
+
+mamba install --update-specs --yes --quiet "conda-forge-ci-setup=3" conda-build pip boa -c conda-forge
+mamba update --update-specs --yes --quiet "conda-forge-ci-setup=3" conda-build pip boa -c conda-forge
 
 # set up the condarc
 setup_conda_rc "${FEEDSTOCK_ROOT}" "${RECIPE_ROOT}" "${CONFIG_FILE}"
@@ -56,7 +58,7 @@ if [[ "${BUILD_WITH_CONDA_DEBUG:-0}" == 1 ]]; then
     # Drop into an interactive shell
     /bin/bash
 else
-    conda $BUILD_CMD "${RECIPE_ROOT}" -m "${CI_SUPPORT}/${CONFIG}.yaml" \
+    conda mambabuild "${RECIPE_ROOT}" -m "${CI_SUPPORT}/${CONFIG}.yaml" \
         --suppress-variables ${EXTRA_CB_OPTIONS:-} \
         --clobber-file "${CI_SUPPORT}/clobber_${CONFIG}.yaml"
     ( startgroup "Validating outputs" ) 2> /dev/null

--- a/.scripts/run_osx_build.sh
+++ b/.scripts/run_osx_build.sh
@@ -1,5 +1,7 @@
 #!/usr/bin/env bash
 
+# -*- mode: jinja-shell -*-
+
 source .scripts/logging_utils.sh
 
 set -xe
@@ -9,7 +11,7 @@ MINIFORGE_HOME=${MINIFORGE_HOME:-${HOME}/miniforge3}
 ( startgroup "Installing a fresh version of Miniforge" ) 2> /dev/null
 
 MINIFORGE_URL="https://github.com/conda-forge/miniforge/releases/latest/download"
-MINIFORGE_FILE="Miniforge3-MacOSX-$(uname -m).sh"
+MINIFORGE_FILE="Mambaforge-MacOSX-$(uname -m).sh"
 curl -L -O "${MINIFORGE_URL}/${MINIFORGE_FILE}"
 rm -rf ${MINIFORGE_HOME}
 bash $MINIFORGE_FILE -b -p ${MINIFORGE_HOME}
@@ -18,14 +20,12 @@ bash $MINIFORGE_FILE -b -p ${MINIFORGE_HOME}
 
 ( startgroup "Configuring conda" ) 2> /dev/null
 
-GET_BOA=boa
-BUILD_CMD=mambabuild
-
 source ${MINIFORGE_HOME}/etc/profile.d/conda.sh
 conda activate base
 
 echo -e "\n\nInstalling conda-forge-ci-setup=3 and conda-build."
-conda install -n base --quiet --yes "conda-forge-ci-setup=3" conda-build pip ${GET_BOA:-}
+mamba install -n base --update-specs --quiet --yes "conda-forge-ci-setup=3" conda-build pip boa
+mamba update -n base --update-specs --quiet --yes "conda-forge-ci-setup=3" conda-build pip boa
 
 
 
@@ -59,7 +59,7 @@ if [[ "${HOST_PLATFORM}" != "${BUILD_PLATFORM}" ]]; then
     EXTRA_CB_OPTIONS="${EXTRA_CB_OPTIONS:-} --no-test"
 fi
 
-conda $BUILD_CMD ./recipe -m ./.ci_support/${CONFIG}.yaml --suppress-variables --clobber-file ./.ci_support/clobber_${CONFIG}.yaml ${EXTRA_CB_OPTIONS:-}
+conda mambabuild ./recipe -m ./.ci_support/${CONFIG}.yaml --suppress-variables --clobber-file ./.ci_support/clobber_${CONFIG}.yaml ${EXTRA_CB_OPTIONS:-}
 ( startgroup "Validating outputs" ) 2> /dev/null
 
 validate_recipe_outputs "${FEEDSTOCK_NAME}"

--- a/LICENSE.txt
+++ b/LICENSE.txt
@@ -1,5 +1,5 @@
 BSD 3-clause license
-Copyright (c) 2015-2021, conda-forge contributors
+Copyright (c) 2015-2022, conda-forge contributors
 All rights reserved.
 
 Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:

--- a/README.md
+++ b/README.md
@@ -13,14 +13,14 @@ Development: https://github.com/restic/restic.git
 
 Documentation: https://restic.readthedocs.io/en/stable/index.html
 
-Restic is a modern backup program that can back up your files
-from Linux, BSD, Mac and Windows
-to many different storage types, including self-hosted and online services
-easily, being a single executable that you can run without a server or complex setup
-effectively, only transferring the parts that actually changed in the files you back up
-securely, by careful use of cryptography in every part of the process
-verifiably, enabling you to make sure that your files can be restored when needed
-freely - restic is entirely free to use and completely open source
+Restic is a modern backup program that can back up your files from Linux,
+BSD, Mac and Windows to many different storage types, including self-hosted
+and online services easily, being a single executable that you can run
+without a server or complex setup effectively, only transferring the parts
+that actually changed in the files you back up securely, by careful use of
+cryptography in every part of the process verifiably, enabling you to make
+sure that your files can be restored when needed freely - restic is
+entirely free to use and completely open source
 
 
 Current build status

--- a/recipe/bld.bat
+++ b/recipe/bld.bat
@@ -4,3 +4,9 @@ go run -mod=vendor build.go --enable-cgo
 rem Install binary
 mkdir -p %PREFIX%\bin
 mv restic.exe %PREFIX%\bin\restic.exe
+
+rem Setup and copy licenses of dependencies
+go install github.com/google/go-licenses@latest
+go-licenses save github.com/restic/restic/cmd/restic --save_path=licenses
+mkdir -p %PREFIX%\info
+mv licenses %PREFIX%\info

--- a/recipe/bld.bat
+++ b/recipe/bld.bat
@@ -8,5 +8,3 @@ cp restic.exe %PREFIX%\bin\restic.exe
 rem Setup and copy licenses of dependencies
 go install github.com/google/go-licenses@latest
 go-licenses save github.com/restic/restic/cmd/restic --save_path=licenses
-mkdir -p %PREFIX%\info
-cp -r licenses %PREFIX%\info

--- a/recipe/bld.bat
+++ b/recipe/bld.bat
@@ -8,4 +8,4 @@ cp restic.exe %PREFIX%\bin\restic.exe
 rem Setup and copy licenses of dependencies
 set GOBIN=%PREFIX%\bin
 go install github.com/google/go-licenses@latest
-go-licenses save github.com/restic/restic/cmd/restic --save_path=licenses
+%PREFIX%\bin\go-licenses save github.com/restic/restic/cmd/restic --save_path=licenses

--- a/recipe/bld.bat
+++ b/recipe/bld.bat
@@ -6,5 +6,6 @@ mkdir -p %PREFIX%\bin
 cp restic.exe %PREFIX%\bin\restic.exe
 
 rem Setup and copy licenses of dependencies
+set GOBIN=%PREFIX%\bin
 go install github.com/google/go-licenses@latest
 go-licenses save github.com/restic/restic/cmd/restic --save_path=licenses

--- a/recipe/bld.bat
+++ b/recipe/bld.bat
@@ -3,10 +3,10 @@ go run -mod=vendor build.go --enable-cgo
 
 rem Install binary
 mkdir -p %PREFIX%\bin
-mv restic.exe %PREFIX%\bin\restic.exe
+cp restic.exe %PREFIX%\bin\restic.exe
 
 rem Setup and copy licenses of dependencies
 go install github.com/google/go-licenses@latest
 go-licenses save github.com/restic/restic/cmd/restic --save_path=licenses
 mkdir -p %PREFIX%\info
-mv licenses %PREFIX%\info
+cp -r licenses %PREFIX%\info

--- a/recipe/build.sh
+++ b/recipe/build.sh
@@ -22,5 +22,5 @@ mkdir -p $PREFIX/bin
 mv restic $PREFIX/bin/restic
 
 # Setup and copy licenses of dependencies
-go install github.com/google/go-licenses@latest
+GOBIN=$PREFIX/bin go install github.com/google/go-licenses@latest
 go-licenses save github.com/restic/restic/cmd/restic --save_path=licenses

--- a/recipe/build.sh
+++ b/recipe/build.sh
@@ -25,4 +25,4 @@ mv restic $PREFIX/bin/restic
 go install github.com/google/go-licenses@latest
 go-licenses save github.com/restic/restic/cmd/restic --save_path=licenses
 mkdir -p $PREFIX/info
-mv licenses $PREFIX/info/
+cp -r licenses $PREFIX/info/

--- a/recipe/build.sh
+++ b/recipe/build.sh
@@ -24,5 +24,3 @@ mv restic $PREFIX/bin/restic
 # Setup and copy licenses of dependencies
 go install github.com/google/go-licenses@latest
 go-licenses save github.com/restic/restic/cmd/restic --save_path=licenses
-mkdir -p $PREFIX/info
-cp -r licenses $PREFIX/info/

--- a/recipe/build.sh
+++ b/recipe/build.sh
@@ -22,7 +22,7 @@ mkdir -p $PREFIX/bin
 mv restic $PREFIX/bin/restic
 
 # Setup and copy licenses of dependencies
-go get github.com/google/go-licenses
+go install github.com/google/go-licenses@latest
 go-licenses save github.com/restic/restic/cmd/restic --save_path=licenses
 mkdir -p $PREFIX/info
 mv licenses $PREFIX/info/

--- a/recipe/build.sh
+++ b/recipe/build.sh
@@ -20,3 +20,9 @@ fi
 # Install binary
 mkdir -p $PREFIX/bin
 mv restic $PREFIX/bin/restic
+
+# Setup and copy licenses of dependencies
+go get github.com/google/go-licenses
+go-licenses save github.com/restic/restic/cmd/restic --save_path=licenses
+mkdir -p $PREFIX/info
+mv licenses $PREFIX/info/

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -10,7 +10,7 @@ source:
   sha256: a9c88d5288ce04a6cc78afcda7590d3124966dab3daa9908de9b3e492e2925fb
 
 build:
-  number: 0
+  number: 1
 
 requirements:
   build:

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -25,7 +25,9 @@ test:
 about:
   home: https://restic.net/
   license: BSD-2-Clause
-  license_file: LICENSE
+  license_file:
+    - LICENSE
+    - licenses/
   summary: Restic is a fast and secure backup program.
   description: |
     Restic is a modern backup program that can back up your files from Linux,

